.github/workflows/ci.yml

```diff
@@ -37,3 +44,10 @@ jobs:
         luarocks lint tl-dev-1.rockspec
         make selfbuild
         luarocks test
+
```

Comment:
This is quite a bit of hoopla and repeated code you don't need to do.

If you just use `actions/cache@v4` instead of manually using the save and restore functions separately then you only need to have one `uses:` block (at the point where the restore would be) and a hook takes care of the save automatically. See here for an example in use:

https://github.com/alerque/polytype/blob/master/.github/workflows/deploy.yml#L27-L31